baremaps-server/src/main/resources/geocoder/index.html

```diff
@@ -0,0 +1,136 @@
+    <style>
+        form {
+            margin-bottom: 1rem;
+        }
+
+        td, th {
+            padding: 1rem;
+            border: 1px solid black;
+        }
+
+        table {
+            border-collapse: collapse;
+        }
+    </style>
+</head>
+<body>
+<!-- A form to submit the IP address -->
+<form onsubmit="searchByAddress(event)">
+    <label for="address">Address:</label>
+    <input type="text" id="address" name="address" value=""/>
+    <input type="submit" value="Submit"/>
+</form>
+<!-- The table of retrieved locations -->
+<table id="results"></table>
+<script>
+    function searchByAddress(event) {
+        event.preventDefault();
+
+        // Get the IP address from the form
+        var address = document.getElementById('address').value;
+
+        // Make a query to retrieve the location from the address
+        const request = new XMLHttpRequest();
+        request.open('GET', `http://localhost:9000/api/geocoder?address=${address}`, true);
+
+        // Set request Accept header to application/json
+        request.setRequestHeader('Accept', 'application/json');
+
+        // Display the request result in the map
+        request.onload = function () {
+            if (request.status >= 200 && request.status < 400) {
+                // Success!
+                const response = JSON.parse(request.responseText);
+
+                // Write the result address in a href
+                // Fill the table of geo locations from the resulting geoLocations
+                // Geo locations contain an address, an ipv4Range, a location, a network and a country
+                const table = document.getElementById('results');
+                table.innerHTML = '';
+                // Insert header row
+                table.insertRow().innerHTML =
+                    `<th>#</th>
+                    <th>Score</th>
+                    <th>Url</th>
+                    <th>Name</th>
+                    <th>Ascii name</th>
+                    <th>Alternate names</th>
+                    <th>Latitude</th>
+                    <th>Longitude</th>
+                    <th>Feature Class</th>
+                    <th>feature Code</th>
+                    <th>Country Code</th>
+                    <th>Country Code 2</th>
+                    <th>Admin1Code</th>
+                    <th>Admin2Code</th>
+                    <th>Admin3Code</th>
+                    <th>Admin4Code</th>
+                    <th>Population</th>
+                    <th>Elevation</th>
+                    <th>Dem</th>
+                    <th>Timezone</th>
+                    <th>Modification Date</th>`;
+                for (let i = 0; i < response.results.length; i++) {
+                    const result = response.results[i];
+                    const row = table.insertRow(i + 1);
+                    let pos = 0;
+                    let cell = row.insertCell(pos++);
+                    cell.innerHTML = i + 1;
```

Comment:
You may use a javascript `for (key in object) {` to iterate generate the table and simplify this part of the code.

Reply:
What about this line though `cell.innerHTML = `<a href="http://www.openstreetmap.org/?mlat=${result.data.latitude}&mlon=${result.data.longitude}&zoom=15" target="_blank">OSM</a>`;?

Reply:
a special case manualy added at the end of the table?